x-pack/plugins/observability/public/components/shared/alert_search_bar/types.ts

```diff
@@ -13,6 +17,35 @@ export interface AlertStatusFilterProps {
   onChange: (id: string, value: string) => void;
 }
 
+export interface AlertSearchBarWithUrlSyncProps extends CommonAlertSearchBarProps {
+  urlStorageKey: string;
+}
+
+export interface KibanaDependencies {
+  data: {
+    query: {
+      timefilter: { timefilter: TimefilterContract };
+    };
+  };
+  notifications: { toasts: ToastsStart };
+  triggersActionsUi: {
+    getAlertsSearchBar: (props: AlertsSearchBarProps) => ReactElement<AlertsSearchBarProps>;
+  };
+}
+
+export type ObservabilityAlertSearchBarDependencies = KibanaDependencies;
+
+export interface Services {
+  timeFilterService: TimefilterContract;
+  AlertsSearchBar: (props: AlertsSearchBarProps) => ReactElement<AlertsSearchBarProps>;
```

Comment:
I don't think this level of specificity is necessary for your use case.  Unless you're interacting directly with those props, you can just say, "this is a React component I'm rendering":

```
AlertsSearchBar: () => ReactNode;
```

Reply:
Since I am passing the related props to this component, I expect the component to get `AlertsSearchBarProps`.